### PR TITLE
SecurityCouncilMemberElectionGovernorCountingUpgradeable.votesToWeight refactor

### PR DESCRIPTION
**Pull request by godzillaba**
* remove numerator and denominator in favor of just `fullWeightDuration`
* misc minor things

### CODE REVIEW

File: src/security-council-mgmt/governors/modules/SecurityCouncilMemberElectionGovernorCountingUpgradeable.sol [hunk 1]
```diff
@@ -265,38 +260,28 @@ abstract contract SecurityCouncilMemberElectionGovernorCountingUpgradeable is
...
-            * (blockNumber - decreasingWeightStartBlock) / WAD;
+        uint256 decreasingWeightDuration = endBlock - fullWeightVotingDeadline_;
 
-        if (decreaseAmount >= votes) {
-            return 0;
-        }
+        // slope numerator is -votes, slope denominator is decreasingWeightDuration, delta x is blockNumber - fullWeightVotingDeadline_
+        // y intercept is votes
+        uint256 decreaseAmount =
+            WAD * votes / decreasingWeightDuration * (blockNumber - fullWeightVotingDeadline_) / WAD;
```
Comment (gzeoneth) [inline src/security-council-mgmt/governors/modules/SecurityCouncilMemberElectionGovernorCountingUpgradeable.sol:282]:
```suggestion
            WAD * votes * (blockNumber - fullWeightVotingDeadline_) / decreasingWeightDuration / WAD;
```
should give slightly better precision

File: src/security-council-mgmt/governors/modules/SecurityCouncilMemberElectionGovernorCountingUpgradeable.sol [hunk 2]
```diff
@@ -265,38 +260,28 @@ abstract contract SecurityCouncilMemberElectionGovernorCountingUpgradeable is
...
 
         // slope denominator
-        uint256 decreasingWeightDuration =
-            WAD * _decreasingWeightDurationNumerator / _durationDenominator * duration / WAD;
-
-        // slope numerator is -votes
-
-        uint256 decreaseAmount = WAD * votes / decreasingWeightDuration
-            * (blockNumber - decreasingWeightStartBlock) / WAD;
+        uint256 decreasingWeightDuration = endBlock - fullWeightVotingDeadline_;
```
Comment (gzeoneth) [inline src/security-council-mgmt/governors/modules/SecurityCouncilMemberElectionGovernorCountingUpgradeable.sol:277]: we should guarantee `fullWeightVotingDeadline_ < endBlock` by enforcing `_fullWeightDuration <= votingDuration` on init

Comment (godzillaba) [reply src/security-council-mgmt/governors/modules/SecurityCouncilMemberElectionGovernorCountingUpgradeable.sol:272]: good point, done here cf8d157308c26fef11b95e2046a45f2928ded4b8

Comment (godzillaba) [reply src/security-council-mgmt/governors/modules/SecurityCouncilMemberElectionGovernorCountingUpgradeable.sol:272]: if you think it's better, i could flip these in the governor init and do the require in the counting module init

```
__SecurityCouncilMemberElectionGovernorCounting_init({
    maxNominees: _maxNominees,
    initialFullWeightDuration: _fullWeightDuration
});
__GovernorSettings_init(0, _votingPeriod, 0);
```

**Review by yahgwai — APPROVED** [review]

**Review by gzeoneth — APPROVED** [review]
LGTM

**Pull request merged**